pkg/rtc/participant.go

```diff
@@ -2251,7 +2256,12 @@ func (p *ParticipantImpl) SendDataPacket(dp *livekit.DataPacket, data []byte) er
 	}
 
 	err := p.TransportManager.SendDataPacket(dp, data)
-	if err == nil {
+	if err != nil {
+		if (err == sctp.ErrStreamClosed || err == io.ErrClosedPipe) && p.params.ReconnectOnDataChannelError {
```

Comment:
These are the two errors coming from Pion. We try to send data only after data channel is open. So, any error from Pion is treated as unrecoverable and can issue a full reconnect.